src/legacy/core_plugins/expressions/public/np_ready/public/loader.ts

```diff
@@ -101,19 +121,19 @@ export class ExpressionLoader {
 
   private execute = async (
     expression: string | ExpressionAST,
-    params: IExpressionLoaderParams
+    params?: IExpressionLoaderParams
   ): Promise<Data> => {
     if (this.dataHandler) {
       this.dataHandler.cancel();
     }
     this.dataHandler = execute(expression, params);
-    const data = await this.dataHandler.getData();
-    this.dataSubject.next(data);
-    return data;
+    this.data = await this.dataHandler.getData();
```

Comment:
I would expect the only call to `this.loadingSubject.next()` to live right above the `getData` call here.

Reply:
see comment above